core/src/typecheck/mod.rs

```diff
@@ -2420,7 +2420,39 @@ pub fn subsumption(
     checked: UnifType,
 ) -> Result<(), UnifError> {
     let inferred_inst = instantiate_foralls(state, &mut ctxt, inferred, ForallInst::UnifVar);
-    checked.unify(inferred_inst, state, &ctxt)
+    match (&inferred_inst, &checked) {
+        (
+            UnifType::Concrete {
+                typ: TypeF::Record(rrows),
+                ..
+            },
+            UnifType::Concrete {
+                typ: TypeF::Dict { type_fields, .. },
+                ..
+            },
+        ) => {
+            for row in rrows.iter() {
+                match row {
+                    GenericUnifRecordRowsIteratorItem::Row(a) => {
+                        subsumption(state, ctxt.clone(), a.typ.clone(), *type_fields.clone())?
+                    }
+                    GenericUnifRecordRowsIteratorItem::TailUnifVar { id, .. } => state
+                        .table
+                        .assign_rrows(id, UnifRecordRows::concrete(RecordRowsF::Empty)),
+                    GenericUnifRecordRowsIteratorItem::TailConstant(id) => {
+                        Err(UnifError::WithConst {
+                            var_kind: VarKindDiscriminant::EnumRows,
```

Comment:
```suggestion
                            var_kind: VarKindDiscriminant::RecordRows,
```

Copy-paste leftover I guess, but we are unifying record rows, not enum rows :slightly_smiling_face: